accounts/queries.py

```diff
@@ -114,39 +114,37 @@ def get_child_participation_eligibility(child, study) -> bool:
     Returns:
         bool: Return true if child is eligible based on their prior study participation
     """
-    # Need to import StudyType here due to circular import problems
-    from studies.models import StudyType
 
-    must_have_participated = True
-    must_not_have_participated = True
+    ember_frame_player_id = 1
+    must_have = True
+    must_not = True
 
     # for both must have and must not have participated, ignore responses from internal studies that are empty
+
     if study.must_have_participated.exists():
-        must_have_participated_all = [
-            i
-            for i in study.must_have_participated.all()
-            if child.responses.filter(study=i)
-            .exclude(study__study_type=StudyType.get_ember_frame_player(), sequence=[])
-            .exists()
-        ]
-
-        # only true if response exists for each of the studies in the list
-        must_have_participated = set(must_have_participated_all) == set(
-            study.must_have_participated.all()
+        must_have_count = (
+            child.responses.filter(study__in=study.must_have_participated.all())
+            .exclude(study__study_type_id=ember_frame_player_id, sequence=[])
+            .distinct()
+            .values_list("study")
+            .count()
         )
+        must_have = must_have_count == study.must_have_participated.count()
+
+        if not must_have:
+            return False
 
     if study.must_not_have_participated.exists():
-        must_not_have_participated_all = [
-            i
-            for i in study.must_not_have_participated.all()
-            if child.responses.filter(study=i)
-            .exclude(study__study_type=StudyType.get_ember_frame_player(), sequence=[])
-            .exists()
-        ]
-        # only true if response does NOT exist for any one (or more) of the studies in the list
-        must_not_have_participated = len(must_not_have_participated_all) == 0
-
-    return must_have_participated and must_not_have_participated
+        must_not_have_count = (
+            child.responses.filter(study__in=study.must_not_have_participated.all())
+            .exclude(study__study_type_id=ember_frame_player_id, sequence=[])
```

Comment:
Now that I'm thinking about query speed, is this `exclude()` part slow? Can you fetch the child responses 1x, exclude empty sequences 1x, and *then* calculate the must have/must have nots? My certainty on this is extremely low, ignore if wrong!

Reply:
I don't think it'll make the query slower, because it is only run after the `count()` function and more restrictive filters usually speed up the query's response.  More restrictions equals less data to return.